vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/certs/serving.go

```diff
@@ -245,7 +246,27 @@ func (c *certKeyFileContent) Equals(rhs *certKeyFileContent) bool {
 }
 
 // GetConfigForClient copied from tls.getCertificate
-func (c *runtimeDynamicLoader) GetConfigForClient(*tls.ClientHelloInfo) (*tls.Config, error) {
+func (c *runtimeDynamicLoader) GetConfigForClient(hello *tls.ClientHelloInfo) (*tls.Config, error) {
 	tlsConfigCopy := c.tlsConfig
+
+	// if the client set SNI information, just use our "normal" SNI flow
+	if len(hello.ServerName) > 0 {
```

Comment:
should we fall back to the logic below?

Reply:
i.e. if the server name cert is not found. Now local IPs only count for empty ServerName. This might be ok.

Reply:
> i.e. if the server name cert is not found. Now local IPs only count for empty ServerName. This might be ok.

I think this behavior is ok.  Based on our unit tests, I'm pretty sure the http serving stack honors the IPs correctly and this avoids some amount of custom code